src/connection.rs

```diff
@@ -520,32 +520,64 @@ impl ActualConnection {
     }
 }
 
-pub fn connect(
+fn build_connection(
     connection_info: &ConnectionInfo,
     timeout: Option<Duration>,
 ) -> RedisResult<Connection> {
     let con = ActualConnection::new(&connection_info.addr, timeout)?;
-    let mut rv = Connection {
+    Ok(Connection {
         con,
         parser: Parser::new(),
         db: connection_info.db,
         pubsub: false,
+    })
+}
+
+fn connect_auth(
+    connection_info: &ConnectionInfo,
+    timeout: Option<Duration>,
+) -> RedisResult<Connection> {
+    let mut rv = build_connection(connection_info, timeout)?;
+    let mut command = cmd("AUTH");
+    if let Some(username) = &connection_info.username {
+        command.arg(username);
+    }
+    let passwd = connection_info.passwd.as_ref().unwrap();
+    let err = match command.arg(passwd).query::<Value>(&mut rv) {
+        Ok(_) => return Ok(rv),
```

Comment:
Should this be `Ok(Value::Okay) => return Ok(rv)`? Same as the second AUTH below

Reply:
Well, I have that too in my non-public iteration of patch. However, if I use `Ok(Value::Okay) => return Ok(rv)`
non-exhaustive error shows up.

```
error[E0004]: non-exhaustive patterns: `Ok(Nil)`, `Ok(Int(_))`, `Ok(Data(_))` and 2 more not covered
```

The question is, what to do with other `Ok(...)` patterns? I have decided, that all `Ok ` values are just fine or we can do better, that `Ok(Value::Okay) => return Ok(rv)` and other Ok(...) patters will fail with message

```
fail!((
   ErrorKind::ResponseError,
   "Invalid data returned during AUTH"
)
```